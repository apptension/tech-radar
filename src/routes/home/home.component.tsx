import React from 'react';
import { FormattedMessage } from 'react-intl';

import { Logo } from '../../shared/components/logo';
import RadarImg from '../../images/radar.png';
import RadarMobileImg from '../../images/radar-mobile.png';
import { Background } from '../../shared/components/background';
import { RadarHomeAnimation } from '../../shared/components/radarHomeAnimation';
import { ROUTES } from '../app.constants';
import { ButtonIcon, ButtonSize } from '../../shared/components/button/button.types';
import { Link } from '../../shared/components/link';
import { TitleTag } from '../../shared/components/titleTag';
import { useLastContentfulUpdate } from '../../shared/hooks/useContentfulData/useContentfulData';
import { useMediaQuery } from '../../shared/hooks/useMediaQuery';
import { Breakpoint } from '../../theme/media';
import messages from './home.messages';
import {
  Container,
  Content,
  Description,
  ExploreLinkContainer,
  Header,
  Image,
  LastUpdateInfo,
  LogoWrapper,
  RadarContent,
  TextContent,
} from './home.styles';

export const Home = () => {
  const { matches: isDesktop } = useMediaQuery({ above: Breakpoint.DESKTOP });
  const { matches: isTablet } = useMediaQuery({ above: Breakpoint.TABLET });
  const lastContentfulUpdate = useLastContentfulUpdate();

  return (
    <Container>
      <Background />
      <Header>
        <LogoWrapper>
          {isDesktop ? (
            <Logo />
          ) : (
            <Link to="https://apptension.com" withBorder={false}>
              <Logo full={false} />
            </Link>
          )}
        </LogoWrapper>
        {isDesktop && (
          <Link to="https://apptension.com" icon={ButtonIcon.OUT}>
            <FormattedMessage {...messages.backToMainPageButton} />
          </Link>
        )}
      </Header>
      <Content>
        <TextContent>
          <TitleTag />
          <Description>
            <FormattedMessage {...messages.description} />
          </Description>
          <ExploreLinkContainer>
            <Link to={ROUTES.explore} size={isDesktop ? ButtonSize.LARGE : ButtonSize.REGULAR} icon={ButtonIcon.ARROW}>
              <FormattedMessage {...messages.exploreButton} />
            </Link>
          </ExploreLinkContainer>
        </TextContent>
        <RadarContent>
          {isDesktop ? <RadarHomeAnimation /> : <Image src={isTablet ? RadarImg : RadarMobileImg} />}
        </RadarContent>
      </Content>
      {!!lastContentfulUpdate && isDesktop && <LastUpdateInfo date={lastContentfulUpdate} />}
    </Container>
  );
};
