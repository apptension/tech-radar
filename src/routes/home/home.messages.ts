/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  description: {
    id: 'home.description',
    defaultMessage:
      'The radar will let you explore all of the technologies, methods, and tools used here at Apptension and learn more about them. You will also get to see the technologies we have on our radar (pun intended) and, hopefully, get inspired.',
  },
  exploreButton: {
    id: 'home.exploreButton',
    defaultMessage: 'Explore',
  },
  backToMainPageButton: {
    id: 'home.backToMainPageButton',
    defaultMessage: 'Back to apptension.com',
  },
  goToAdminPanel: {
    id: 'newEntry.goToAdminPanel',
    defaultMessage: 'Go to AdminPanel',
  },
});
