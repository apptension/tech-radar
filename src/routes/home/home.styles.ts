import styled, { css, keyframes } from 'styled-components';

import { Paragraph } from '../../theme/typography';
import { Header as HeaderComponent } from '../../shared/components/header';
import { TitleTag as TitleTagComponent } from '../../shared/components/titleTag';
import { LastUpdateInfo as LastUpdateInfoComponent } from '../../shared/components/lastUpdateInfo';
import { mediaQuery } from '../../theme/media';
import { zIndex } from '../../theme';

export const Container = styled.div`
  height: 100%;
  width: 100%;
  position: relative;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
`;

export const Header = styled(HeaderComponent)`
  display: flex;
  justify-content: center;

  ${mediaQuery.desktop} {
    justify-content: space-between;
  }
`;

export const LogoWrapper = styled.div`
  height: 50px;
`;

export const Content = styled.div`
  width: 100%;
  height: 100%;
  position: relative;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: flex-start;
  padding-top: 110px;
  max-width: 1500px;
  margin: 0 auto;
`;

export const TextContent = styled.div`
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: flex-start;
  margin-bottom: 56px;
  text-align: center;
  max-width: 722px;
`;

export const textEnterAnimation = keyframes`
  0% {opacity: 0; }
  100% {opacity: 1; filter: blur(0); transform: translateY(0)}
`;

export const textAnimationStyles = css`
  opacity: 0;
  filter: blur(4px);
  transform: translateY(-25px);
  animation-name: ${textEnterAnimation};
  animation-duration: 2.3s;
  animation-timing-function: ease-in-out;
  animation-fill-mode: forwards;
  z-index: ${zIndex.contentOverlay};
`;

export const TitleTag = styled(TitleTagComponent)`
  ${textAnimationStyles}
`;

export const Description = styled(Paragraph)`
  z-index: ${zIndex.header};
  animation-delay: 0.23s;
  margin: 24px 24px 36px;
  ${textAnimationStyles}

  ${mediaQuery.desktop} {
    margin: 48px 0;
  }
`;

export const ExploreLinkContainer = styled.span`
  z-index: ${zIndex.header};
  animation-delay: 0.7s;
  ${textAnimationStyles}
`;

const radarContentStyles = css`
  flex: 1;
  display: flex;
  justify-content: center;
  position: absolute;
`;

export const RadarAnimationContainer = styled.div`
  align-items: flex-end;
  bottom: 0;
  ${radarContentStyles}
`;

export const RadarContent = styled.div`
  align-items: flex-end;
  max-width: 80vw;
  bottom: -150px;
  ${radarContentStyles}
`;

export const MobileRadarContent = styled.div<{ moveImageBy: number }>`
  align-items: flex-start;
  bottom: ${(props) => props.moveImageBy}px;
  max-height: 60%;
  ${radarContentStyles}
`;

export const Image = styled.img`
  width: 100vw;
  object-fit: contain;

  ${mediaQuery.desktop} {
    width: 100%;
  }
`;

export const LastUpdateInfo = styled(LastUpdateInfoComponent)`
  position: absolute;
  bottom: 24px;
  right: 35px;
`;
