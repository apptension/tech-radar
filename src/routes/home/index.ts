export { Home } from './home.component';
