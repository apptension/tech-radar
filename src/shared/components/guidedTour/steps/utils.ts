import { color } from '../../../../theme';
import { highlightBlip, unhighlightBlip } from '../../../utils/radarUtils';
import { RadarTechnology } from './../../radar/radar.types';

export const highlightBlips = (list: RadarTechnology[]) => {
  list.forEach((tech) => {
    highlightBlip({ id: tech?.id || '', ring: Number(tech?.ring), scale: false });
  });
};

export const unHighlightBlips = (list: RadarTechnology[]) => {
  list.forEach((tech) => {
    unhighlightBlip({
      id: tech?.id || '',
      ring: Number(tech?.ring),
      color: color.mineShaft,
    });
  });
};
