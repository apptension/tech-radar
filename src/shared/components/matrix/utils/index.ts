import { Skill } from '../types';

export const checkIfSkillIsAdded = (skills: Skill[], skillValue: string) =>
  skillValue !== skills.find((item) => item.value === skillValue)?.value;

export const findSkill = (skillValue: string, skills: Skill[]) => {
  return skills.find((skill) => skill.value === skillValue) as Skill;
};

export const mapSkillToSkillWithVisibility = (skill: Skill) => ({ ...skill, isVisible: true });

export const getUserSkillsFromIds = (skillIds: string[], skills: Skill[]) =>
  skillIds.map((skillId) => {
    const skill = findSkill(skillId, skills);
    return mapSkillToSkillWithVisibility(skill);
  });

export const mapSkillsToValues = (skills: Skill[]) => skills.map(({ value }) => value);
