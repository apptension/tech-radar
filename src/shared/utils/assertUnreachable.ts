export const assertUnreachable = (value: never, message: string): never => {
  throw new Error(message);
};
