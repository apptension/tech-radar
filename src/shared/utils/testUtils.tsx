import React, { ReactNode, ReactElement } from 'react';
import { HelmetProvider } from 'react-helmet-async';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { identity } from 'ramda';
import { Provider } from 'react-redux';
import { createMemoryHistory, MemoryHistory } from 'history';
import { Route, Router } from 'react-router';
import { IntlProvider } from 'react-intl';
import produce from 'immer';

import { DEFAULT_LOCALE, translationMessages, MessagesObject } from '../../i18n';
import { store as fixturesStore } from '../../mocks/store';
import createReducer, { GlobalState } from '../../config/reducers';

export const PLACEHOLDER_TEST_ID = 'content';
export const PLACEHOLDER_CONTENT = <span data-testid="content">content</span>;

export const spiedHistory = (route = '/') => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const pushSpy = jest.spyOn(history, 'push');
  return {
    history,
    pushSpy,
  };
};

interface ContextData {
  router?: {
    url?: string;
    routePath?: string;
    history?: MemoryHistory;
  };
  store?: GlobalState;
  messages?: MessagesObject;
}

interface ProvidersWrapperProps {
  children: ReactNode;
  context?: ContextData;
}

export const ProvidersWrapper = ({ children, context = {} }: ProvidersWrapperProps) => {
  const { router = {}, store = fixturesStore, messages } = context;
  const { url = `/${DEFAULT_LOCALE}`, routePath = '/:lang/', history } = router;

  const routerHistory: MemoryHistory = history ?? createMemoryHistory({ initialEntries: [url] });

  const intlProviderMockProps = {
    locale: DEFAULT_LOCALE,
    messages: messages ?? translationMessages[DEFAULT_LOCALE],
  };

  return (
    <Router history={routerHistory}>
      <HelmetProvider>
        <IntlProvider {...intlProviderMockProps}>
          <Provider store={createStore(createReducer(), produce(store, identity))}>
            <Route path={routePath}>{children}</Route>
          </Provider>
        </IntlProvider>
      </HelmetProvider>
    </Router>
  );
};

export const makeContextRenderer = <T, _>(component: (props: T | Record<string, never>) => ReactElement) => (
  props?: T,
  context?: ContextData
) =>
  render(component(props ?? {}), {
    wrapper: ({ children }) => <ProvidersWrapper context={context}>{children}</ProvidersWrapper>,
  });

export const makePropsRenderer = <T, _>(component: (props: T | Record<string, never>) => ReactElement) => (props?: T) =>
  render(component(props ?? {}));
