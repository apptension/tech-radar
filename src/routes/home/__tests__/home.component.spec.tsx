import React from 'react';
import { Home } from '../home.component';
import { makeContextRenderer } from '../../../shared/utils/testUtils';

describe('Home: Component', () => {
  const component = () => <Home />;
  const render = makeContextRenderer(component);

  it('should render correctly', () => {
    const { container } = render();
    expect(container.firstChild).toMatchSnapshot();
  });
});
