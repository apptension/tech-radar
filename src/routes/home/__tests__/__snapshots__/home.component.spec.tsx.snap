// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home: Component should render correctly 1`] = `
.c1 {
  font-family: OpenSans;
  font-weight: bold;
  color: #000000;
}

.c3 {
  padding: 8px 16px;
  border: 1px solid #80809B;
  color: #42f272;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c2 {
  width: 100px;
  margin-bottom: 20px;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Hello world!
  </h1>
  <svg
    class="c2"
  >
    logo.svg
  </svg>
  <select
    class=""
  >
    <option
      value="en"
    >
      en
    </option>
    <option
      value="pl"
    >
      pl
    </option>
  </select>
  <div
    class=""
  >
    <button
      class="c3"
    >
      Fetch users
    </button>
    <ul />
  </div>
</div>
`;
